File: python/122-best-time-to-buy-and-sell-stock-II.py
class Solution:
    # @param {integer[]} prices
    # @return {integer}
    def maxProfit(self, prices):
        l = len(prices)
        res = [0] * l
        if l == 0 or l == 1:
            return 0
        for i in range(1, l):
            diff = prices[i] - prices[i-1]
            res[i] = diff + res[i-1] if diff > 0 else res[i-1]
        return res[l-1]

File: javascript/src/122-best-time-to-buy-and-sell-stock-II.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    var diff, len = prices.length, res = [];
    if (!len) {
        return 0;
    }
    res[0] = 0;
    for (var i=1; i < len; i++) {
        diff = prices[i] - prices[i-1];
        res[i] = diff > 0 ? (res[i-1] + diff) : res[i-1];
    }
    return res[len-1];
};